src/librustc_traits/chalk/lowering.rs

```diff
@@ -78,10 +78,12 @@ impl<'tcx> LowerInto<'tcx, chalk_ir::InEnvironment<chalk_ir::Goal<RustInterner<'
     ) -> chalk_ir::InEnvironment<chalk_ir::Goal<RustInterner<'tcx>>> {
         let clauses = self.environment.into_iter().filter_map(|clause| match clause {
             ChalkEnvironmentClause::Predicate(predicate) => {
-                match predicate.kind() {
-                    ty::PredicateKind::Trait(predicate, _) => {
+                // FIXME(chalk): forall
```

Comment:
what is this fixme :)

Reply:
oh, I guess I see, the point is that we could actually clean this up to be much nicer, right? at least in theory

Reply:
i think we should be able to, considering that this change gets us closer to how chalk works